numpy/testing/tests/test_utils.py

```diff
@@ -289,17 +290,19 @@ def test_error_message(self):
         except AssertionError as e:
             msg = str(e)
             msg2 = msg.replace("shapes (2L,), (1L, 2L)", "shapes (2,), (1, 2)")
-            msg_reference = "\nArrays are not equal\n\n" \
-                "(shapes (2,), (1, 2) mismatch)\n" \
-                " x: array([1, 2])\n" \
-                " y: [repr failed for <matrix>: The truth value of an array " \
-                "with more than one element is ambiguous. Use a.any() or " \
-                "a.all()]"
+            msg_reference = textwrap.dedent("""\
+
+            Arrays are not equal
+
+            (shapes (2,), (1, 2) mismatch)
+             x: array([1, 2])
+             y: [repr failed for <matrix>: %d format: a number is required, not matrix]""")
```

Comment:
Nothing like having tests that require things to fail in a very specific way...

Reply:
This test is a bit weird also. It should really say `y: matrix([[1,2]])`.

The repr fails because `build_err_msg` uses `np.array_repr(mat))` instead of just `repr(mat)`. Not sure if the bug is in `build_err_msg` or in `np.matrix`.

In any case that's unrelated to this PR.

Reply:
Right, this is #8442. I think the easiest fix would be to rewrite `array_repr` to use full-indices, `[i, j, k]`, rather than recursive `[i][j][k]`.